bandersnatch/precomp_multiexp.go

```diff
@@ -75,52 +73,37 @@ func DeserializePrecomputedLagrange(reader io.Reader) (*PrecomputeLagrange, erro
 		pcl.inner[i].matrix = make([]PointAffine, rowLen)
 		for j := int64(0); j < rowLen; j++ {
 
-			pcl.inner[i].matrix[j] = *ReadUncompressedPoint(reader)
+			pcl.inner[i].matrix[j] = ReadUncompressedPoint(reader)
 		}
 	}
 
 	return &pcl, nil
 }
 
 func (p *PrecomputeLagrange) Commit(evaluations []fr.Element) *PointAffine {
+	var result PointProj
+	result.Identity()
 
-	nbTasks := runtime.NumCPU()
-	chPartialResults := make(chan PointProj, nbTasks)
-	parallel.Execute(len(evaluations), func(start, end int) {
```

Comment:
Interesting, why did you remove the parallel code here? Did you see a performance degradation?

Reply:
The `parallel.Execute` has some unfortunate heap allocations due to the nature of spinning up goroutines and work partitioning. First, I tried to see if that could be avoided but wasn't really possible.

But then I simply challenged the assumption that this work really needed parallel work. So took some benchmarks about how heavy was each operation and was not big enough to justify the overhead of partitioning and goroutines.

The unparalleled execution turned out to be faster, and avoid the CPU overhead plus indirect allocations of the parallelization.

Reply:
Yeah it makes sense that for small multi exponentations (256), the parallel code might incur a non-negligble overhead